### PR TITLE
ci: Use Cirrus CI dockerfile env

### PR DESCRIPTION
**Pull request by maflcko**
Currently the CI env has many intermittent issues:

* The Ubuntu package servers are frequently down
* Occasionally other stuff is down, such as dnf, pip, or the android sdk
* Installing packages is slower than downloading them, at least on Cirrus, which has a fast download speed

Fix all issues by using the Cirrus CI dockerfile env.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -210,6 +210,7 @@ task:
     docker_arguments:
       CI_IMAGE_NAME_TAG: ubuntu:jammy
       FILE_ENV: "./ci/test/00_setup_env_win64.sh"
+  << : *CREDITS_TEMPLATE
```
Comment (josibake) [inline .cirrus.yml:213]: this is unrelated to switching to the docker file, right? would prefer this change in it's own PR because I have a few questions about cost (before / after, etc), but not a super big deal

**Review by josibake — COMMENTED** [review]
code review ACK https://github.com/bitcoin/bitcoin/pull/27340/commits/fa4a46de0b3c1a5895e95dba7e95278932fbfc2c

read through the linked documentation from cirrus and this definitely makes sense. I tried to run the ci locally using `cirrus run` but ran into some issues (unrelated to this PR). If you have any tips on how to test this, happy to give it a go, but the code looks good otherwise.

**Comment by josibake** [comment]
> I can help with running the CI locally, but not how to run Cirrus locally. I don't think testing this is possible, other than looking at the CI tasks for this pull?

I installed the cirrus cli tool and ran `cirrus run` in the directory, which then picks up the `.cirrus.yml` file and executes with cirrus workers, but was running into a bunch of permissions issues. Not related to this PR, so I'm not going to try and troubleshoot it further

**Comment by fanquake** [comment]
Going to merge this now, so that `25.x` and `master` will have the same CI infra.
